environment-dev.yml

```diff
@@ -0,0 +1,13 @@
+name: pyogrio-dev
+channels:
+  - conda-forge
+dependencies:
+  - cython
+  - numpy
+  - gdal ==3.7.0
```

Comment:
Can`gdal` be dropped and just use `libgdal==3.5.1`?  (only picked this version to match `ci/environment-libgdal3.5.1.yml`)

Other than our CI tests for `osgeo/gdal:ubuntu-small-latest`, we're not yet specifically testing GDAL > 3.6.2.  Though when we bump the versions we're testing, we should also do so in this environment...

Reply:
gdal==3.5.1 seemed to give some weird issues in combination with pyarrow, but gdal 3.6.2 seems to work fine.

I also added pre-commit and mentioned in the general develop that it is recommended to install pre-commit. Feedback welcome.